xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-default/src/main/java/org/xwiki/icon/internal/DefaultIconManager.java

```diff
@@ -91,6 +93,33 @@ public String renderHTML(String iconName, String iconSetName, boolean fallback)
         return iconRenderer.renderHTML(iconName, iconSet);
     }
 
+    @Override
+    public Map<String, Object> getMetaData(String iconName) throws IconException
+    {
+        return this.getMetaData(iconName, getIconSet(iconName).getName());
```

Comment:
This means ``getIconSet`` will be called twice for the most common use case. I would add a private ``getMetaData(String iconName, IconSet iconSet)`` and call it from the rest of the ``getMetaData()`` methods.